### PR TITLE
Editor

### PR DESCRIPTION
**Pull request by marcusellison**
React_App

- Named container nodes in plasmic (if not named) to make node accessible in generated container component
- Installed slate editor package with peer dependencies
- created slate editor read and compose components
- adjusted default styling

### CODE REVIEW

File: src/component/editor/compose/index.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+      value={value}
+      onChange={(newValue: any) => {
+        setValue(newValue as SlateDocument);
+
+        //trigger mention
+        onChangeMention(editor);
+
+        //save to local storage for to persist...
+        const content = JSON.stringify(value);
+        localStorage.setItem("content", content);
```
Comment (xh3b4sd) [inline src/component/editor/compose/index.tsx:127]: I would like to better understand what this local storage operation is about.

File: src/component/editor/config/data.js [hunk 2]
```diff
@@ -0,0 +1,405 @@
+export const CHARACTERS = [
```
Comment (xh3b4sd) [inline src/component/editor/config/data.js:1]: This is a `.js` file and I thought we prohibit that.

**Review by xh3b4sd — CHANGES_REQUESTED** [review]
Here are a couple of things I found. Most critical one is where we actually left off with the old react app. I can still break the whole thing when playing around with backticks. I type a backtick character, remove it, and repeat until the whole page is broken. 

Screenshot 2020-11-19 at 13 27 09



---



The blue arrow shows that I cannot move my cursor below the code block element. Usually I would press "arrow down" and get to the next line. The red arrow shows that the code block design goes beyond the action bar. That looks a little odd. 

Screenshot 2020-11-19 at 13 21 37



---



Since we try to do small iterations I was experiencing something unexpected with the content of the updates. Not a big thing, though this has nothing to do with the desired changes of this PR and caused friction on my end where I am not sure about the reasons and expectations. 

Screenshot 2020-11-19 at 13 22 55

**Comment by marcusellison** [comment]
What could make sense here is to reduce scope of initial editor so that +/- is more manageable 🤔

File: src/component/editor/config/data.js (hunk 2 above)
Comment (marcusellison) [reply src/component/editor/config/data.js:1]: resolved

File: src/component/editor/compose/index.tsx (hunk 1 above)
Comment (marcusellison) [reply src/component/editor/compose/index.tsx:104]: As someone types into the editor, we save the state in localStorage aka web storage. This should mean that if someone refreshes the page in the middle of typing a message we save the text they've already written in localStorage.

This isn't enable yet. I'll make a new pull request to enable.

**Comment by marcusellison** [comment]
I'll chain further adjustments to new pull requests to keep changes small, isolated.

**Comment by xh3b4sd** [comment]
Which of the PRs you created is now the latest state I should be testing myself?

**Pull request merged**